npm/cypress-schematic/src/builders/cypress/index.ts

```diff
@@ -99,8 +99,20 @@ export function startDevServer ({
   devServerTarget,
   watch,
   context }: StartDevServerProps): Observable<string> {
-  const overrides = {
-    watch,
+  // @NOTE: Do not forward watch option if not supported by the target dev server,
+  // this is relevant for running Cypress against dev server target that does not support this option,
+  // for instance @nguniversal/builders:ssr-dev-server.
+  // see https://github.com/nrwl/nx/blob/f930117ed6ab13dccc40725c7e9551be081cc83d/packages/cypress/src/executors/cypress/cypress.impl.ts
+  const angularJson = require(`${context.workspaceRoot }/angular.json`)
```

Comment:
Rather than read the config, can you try utilizing `context.getBuilderNameForTarget(targetFromTargetString(devServerTarget))`? That should give you the executor you need.